pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -275,6 +275,14 @@ func (r *ReconcileUserSignup) provisionMasterUserRecord(userSignup *toolchainv1a
 	}
 
 	logger.Info("Created MasterUserRecord", "Name", userSignup.Name, "TargetCluster", targetCluster)
+
+	// If the MasterUserRecord was created without error, update the status to set Complete to true
+	statusError := r.setStatusComplete(userSignup, "")
```

Comment:
No, no, you can't do that here. You just have requested the API server to create a MUR. You can't update the status yet. The creation (actual) of MUR will trigger another reconcile (MUR is watched by the usersignup controller as a secondary resource). And in that following reconcile the status will be set to complete (in the very beginning of the function where you check if the corresponding MUR already exists).

Reply:
This workflow has been fixed.